src/twisted/application/app.py

```diff
@@ -681,3 +690,17 @@ def startApplication(application, save):
         reactor.addSystemEventTrigger('after', 'shutdown', p.save, 'shutdown')
     reactor.addSystemEventTrigger('before', 'shutdown',
                                   service.IService(application).stopService)
+
+
+
+def exitWithSignal(sig):
+    """
+    Force the application to terminate with the specified signal by replacing
+    the signal handler with the default and sending the signal to ourselves.
+
+    @param sig:  Signal to use to terminate the process with C{os.kill}.
+    @type sig:  C{int}
+
+    """
+    signal.signal(sig, signal.SIG_DFL)
```

Comment:
@glyph  Yes but I don't think that is a valid thing to do in all cases?.  If setting the exit status to 128 + signal was always the right thing to do on all platforms, this would (I think) work.  I actually went this route initially, but changed after reading some of the referenced articles in the original ticket.  My concern is that, while the GNU C library certainly obeys the above behavior, I haven't found any spec/guarantee to indicate this would be consistent with other implementations.  Thus, rather than assume it is ok, I chose to let the OS do what it would normally do had the signal not been caught in the first place.  Comment 8 in the ticket, and the linked article talk a little about calling exit.

In approaching this bug I looked for some consistent, authoritive reference on valid exit status', and realized it is somewhat of a platform/implementation specific mess.  Therefore, lacking a better alternative, I wanted to let the OS take control and do what it would normally do.